### ISSUE TITLE
Train on all data [ blech_bot ]

### ISSUE THREAD

**Issue by abuzarmahmood** [abuzarmahmood/blechRNN#5]
No need to cross validate... Model will not be used for out of sample prediction

**Comment by abuzarmahmood** [comment]
The issue involves modifying the code to train on all data without performing cross-validation. The following files and changes were suggested:

1. **File: `run_model.py`**
   - **Remove Train-Test Split:**
     - Lines 116-127: The code currently splits the data into training and testing sets. This split should be removed to train on all data.
   - **Modify Training Call:**
     - Lines 151-161: The `train_model` function is called with `test_inputs` and `test_labels`. These should be removed to avoid cross-validation.

   Suggested code changes:
   ```python
   # Remove train-test split
   # train_test_split = 0.75
   # train_inds = np.random.choice(
   #         np.arange(inputs.shape[1]), 
   #         int(train_test_split * inputs.shape[1]), 
   #         replace = False)
   # test_inds = np.setdiff1d(np.arange(inputs.shape[1]), train_inds)

   # Use all data for training
   train_inputs = inputs.to(device)
   train_labels = labels.to(device)

   # Modify the train_model call
   net, loss, _ = train_model(
           net, 
           train_inputs, 
           train_labels, 
           output_size=output_size,
           lr=0.001, 
           train_steps=15000,
           loss=loss_name
           )
   ```

2. **File: `train.py`**
   - **Remove Cross-Validation Logic:**
     - Lines 61-64: The `cross_val_bool` logic checks if cross-validation should be performed. This should be removed or set to `False`.
   - **Remove Cross-Validation Loss Calculation:**
     - Lines 88-96: The code calculates cross-validation loss every 100 steps. This should be removed.

   Suggested code changes:
   ```python
   # Remove cross-validation logic
   # cross_val_bool = np.logical_and(
   #         test_inputs is not None, 
   #         test_labels is not None
   #         )

   # Remove cross-validation loss calculation
   # if cross_val_bool and (i % 100 == 99): 
   #     test_out, _ = net(test_inputs)
   #     test_out = test_out.reshape(-1, output_size)
   #     test_labels = test_labels.reshape(-1, output_size)
   #     test_loss = criterion(test_out, test_labels)
   #     cross_val_loss[i] = test_loss.item()
   #     cross_str = f'Cross Val Loss: {test_loss.item():0.4f}'
   # else:
   #     cross_str = ''
   ```

These changes will ensure that the model is trained on the entire dataset without performing cross-validation, as requested.

---
*This response was automatically generated by blech_bot*

**Comment by abuzarmahmood** [comment]
But, cross-val loss is still useful to prevent over-fitting.
Closing issue.
